docs/rules/no-whitespace-before-property.md

```diff
@@ -0,0 +1,66 @@
+# Disallow whitespace before properties (no-whitespace-before-property)
+
+JavaScript allows whitespace between objects and their properties. However, inconsistent spacing can make code harder to read and can lead to errors.
+
+```js
+foo. bar .baz . quz
+```
+
+## Rule Details
+
+This rule alerts for whitespace around the dot or brackets before properties of objects if they are on the same line. It does not alert for whitespace when the object and property are on separate lines, as it is common to add newlines to longer chains of properties to improve readability:
```

Comment:
"Readability" is personal and not everyone agrees on what is readable. I'd just drop that part from the sentence.